### PR TITLE
Improve LocalConstantFinder performance

### PR DESCRIPTION
**Pull request by staabm**
Do cheaper checks first. Similar to https://github.com/rectorphp/rector-src/pull/3651

### CODE REVIEW

File: src/PhpParser/NodeFinder/LocalConstantFinder.php [hunk 1]
```diff
@@ -28,21 +28,22 @@ public function match(ClassConstFetch $classConstFetch): ?Const_
             return null;
         }
 
-        $constantClassType = $this->nodeTypeResolver->getType($classConstFetch->class);
-        if (! $constantClassType instanceof TypeWithClassName) {
+        $constatName = $this->nodeNameResolver->getName($classConstFetch->name);
+        if ($constatName === null) {
             return null;
         }
+        $const = $this->findConstantByName($class, $constatName);
```
Comment (samsonasik) [inline src/PhpParser/NodeFinder/LocalConstantFinder.php]: this can be moved down to last statement to improve performance

Comment (staabm) [reply src/PhpParser/NodeFinder/LocalConstantFinder.php]: good catch!

**Review by samsonasik — APPROVED** [review]

**Comment by samsonasik** [comment]
Thank you @staabm

**Comment by samsonasik** [comment]
not sure why test randomly crash

**Comment by staabm** [comment]
> not sure why test randomly crash

I have no idea either. looks like a bug in php-src or even deeper

**Pull request merged**